engine/src/main/java/org/terasology/world/block/BlockRegions.java

```diff
@@ -16,98 +17,134 @@
 import java.util.stream.StreamSupport;
 
 public final class BlockRegions {
+    /**
+     * An invalid block region.
+     * <p>
+     * CAUTION: Behavior for this region may be undeterministic for some operations. Avoid extensive use and encode an
+     * <i>empty region</i> by other means, e.g., by using {@code null} or {@link java.util.Optional}.
+     * <p>
+     * This region may be used as initial value for reductions on block regions. For instance, to compute the union (the
+     * region that encompasses all regions in some collection {@code regions}) the following Stream-API snippet can be
+     * used:
+     * <pre>
+     *     BlockRegion union = regions.stream().reduce(BlockRegions.INVALID, BlockRegion::union, BlockRegion::union);
+     * </pre>
+     */
+    public static final BlockRegion INVALID = new BlockRegion();
```

Comment:
I've asked about this on Discord: https://discord.com/channels/270264625419911192/701818347238588561/790546593911799808

Reply:
Note: this either has to be a `Supplier<BlockRegion>` or should return a `BlockRegionc` so that it cannot be modified.